admin/server/users.go

```diff
@@ -284,6 +284,76 @@ func (s *Server) GetUser(ctx context.Context, req *adminv1.GetUserRequest) (*adm
 	return &adminv1.GetUserResponse{User: userToPB(user)}, nil
 }
 
+func (s *Server) DeleteUser(ctx context.Context, req *adminv1.DeleteUserRequest) (*adminv1.DeleteUserResponse, error) {
+	observability.AddRequestAttributes(
+		ctx,
+		attribute.String("args.email", req.Email),
+		attribute.String("args.org", req.Organization),
+	)
+
+	org, err := s.admin.DB.FindOrganizationByName(ctx, req.Organization)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	role, err := s.admin.DB.FindOrganizationRole(ctx, database.OrganizationRoleNameAdmin)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	currentUser, err := s.GetCurrentUser(ctx, &adminv1.GetCurrentUserRequest{})
+	if err != nil {
+		return nil, err
+	}
+
+	claims := auth.GetClaims(ctx)
+
+	isCurrentUser := currentUser.User.Email == req.Email
+	isSuperuser := claims.Superuser(ctx)
+
+	if !isCurrentUser && !isSuperuser {
+		return nil, status.Error(codes.PermissionDenied, "only superusers can delete other users")
+	}
+
+	orgAdminUsers, err := s.admin.DB.FindOrganizationMemberUsersByRole(ctx, org.ID, role.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(orgAdminUsers) == 1 && orgAdminUsers[0].Email == req.Email {
+		return nil, status.Error(codes.PermissionDenied, "cannot delete the last admin of an organization, please transfer ownership first")
+	}
+
+	memberCount, err := s.admin.DB.CountMembersByOrganization(ctx, org.ID)
+	if err != nil {
+		return nil, err
+	}
+	if memberCount == 1 {
+		// TODO: Tear down the org and all its projects
```

Comment:
We need to implement this because many users will have created their own org and deployed test projects (especially the ones Himadri is looking to clean up). But if you prefer, we can merge this PR and do it in a follow up.

Reply:
Nah no need, just something I wanted clarity